src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -244,7 +253,8 @@ private boolean executeCmd(List<String> cmds) throws IOException, InterruptedExc
         }
 
         private Result executeCmd(String cmd) throws IOException, InterruptedException {
-            PrintStream stdout = listener.getLogger();
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            TeeOutputStream stdout = new TeeOutputStream(listener.getLogger(), baos);
```

Comment:
Does `TeeOutputStream` propagate close to both child streams, because if it does then you would need to wrap `listener.getLogger()` to protect it from being closed when you close `stdout`